plugins/modules/ec2_vpc_igw.py

```diff
@@ -152,6 +153,32 @@ def get_matching_igw(self, vpc_id):
 
         return igw
 
+    def wait_for_igw(self, vpc_id):
+        """
+        Waits for existing igw to be returned via describe_internet_gateways
+        in get_matching_igw
+        :param vpc_id: VPC's ID
+        :return success: True if found, False if not
+        :return igw: igw found
+        """
+        polling_increment_secs = 10
+        max_retries = 10
+        success = False
+
+        for x in range(0, max_retries):
```

Comment:
This is OK, but I would recommend you to look at [Exponential Backoff](https://en.wikipedia.org/wiki/Exponential_backoff). tl;dr - See this implementation - https://github.com/ansible-collections/community.vmware/blob/e31483fcd11e1a05bfc12cb1913bf2d26483903b/plugins/module_utils/vmware.py#L155 

No blocker as such.

Reply:
replaced the manual waiter with module_util waiter, but will definitely keep this in mind moving forward! Thanks @Akasurde